mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -150,7 +153,7 @@ def __init__(
             self.optimizer_parameter_space,
             objectives=self._optimization_targets,
             name=run_name,
-            output_directory=Path(output_directory),
+            output_directory=Path(output_directory) if output_directory else Path("smac3_output"),
```

Comment:
This shouldn't happen, right?
(we set it to a temporary path above)

Please revert - seems unrelated.

Reply:
This is to fix a linter error that slipped through a pervious PR. It won't  pass linting otherwise

Reply:
Add an `assert output_directory` above this instead then.

Reply:
(sometimes the linter logic changes)